jetty-core/jetty-io/src/test/java/org/eclipse/jetty/io/IOResourcesTest.java

```diff
@@ -46,12 +50,88 @@ public void tearDown()
         assertThat("Leaks: " + bufferPool.dumpLeaks(), bufferPool.getLeaks().size(), is(0));
     }
 
-    public static Stream<Resource> all()
+    private static class TestContentSourceFactoryResource extends Resource implements Content.Source.Factory
     {
-        URI resourceUri = MavenTestingUtils.getTestResourcePath("keystore.p12").toUri();
+        private final URI uri;
+        private final ByteBuffer buffer;
+
+        public TestContentSourceFactoryResource(URI uri, byte[] bytes)
+        {
+            this.uri = uri;
+            this.buffer = ByteBuffer.wrap(bytes);
+        }
+
+        @Override
+        public boolean exists()
+        {
+            return true;
+        }
+
+        @Override
+        public long length()
+        {
+            return buffer.remaining();
+        }
+
+        @Override
+        public Path getPath()
+        {
+            return null;
```

Comment:
One should be able to set the Path via constructor, so we can also test this use case (that I mentioned above).

Reply:
This resource only needs to implement `Content.Source.Factory`; testing a resource that has a `Path` already is covered by the 1st parameter returned by the `all()` method.

Reply:
No, what I mean is we need to be able to test that `Content.Source.Factory` is used for a resource implementing this _even if that resource returns some path_ via `getPath`. Right now, we don't test that.

Testing this will prevent against unexpected reordering of all these special cases in `IOResources. asContentSource`.

Reply:
I've changed `getPath()` and `newInputStream()` to make them throw, so the tests will fail if those are called before the `instanceof Content.Source.Factory` check is made.